src/components/DownloadButton.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <a
+    download
+    :href="href"
+    class="btn btn-primary"
+    :class="computedClasses"
+    @click="setLoadingState"
+  >
+    <span
+      v-if="isLoading"
+      class="spinner-border spinner-border-sm"
+      role="status"
+      aria-hidden="true"
+    ></span>
+    <span v-if="isLoading"> Loading...</span>
+    <span v-else> <slot /></span>
+  </a>
+</template>
+
+<script>
+export default {
+  name: 'DownloadButton',
+  props: {
+    href: String,
+    customClass: String,
+    classes: Object,
+    disabled: Boolean
+  },
+  mounted () {
+    window.addEventListener('focus', this.clearLoadingState)
+  },
+  destroyed () {
+    window.addEventListener('focus', this.clearLoadingState)
+  },
```

Comment:
interesting. i wonder if adding a comment in the code, explaining the rationale in the PR re: detecting complete downloads, might be useful going forward?

Reply:
Oh yeah, that's a good idea.